nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -379,8 +379,10 @@ def validate(self, loader=None):
                 model=self.model,
                 loader=loader,
                 device=self.device,
-                criterion=self.loss_function,
+                criterion=self.error_loss,
+                complexity_loss_fn=self.complexity_loss,
                 batches_in_epoch=self.batches_in_epoch,
+                to_device_fn=self.send_data_to_device,
```

Comment:
`to_device_fn` name is not clear to me. The name implies it is only used to transform from one device to another.
But from what I understood this callback is also used to apply final transformations to the model results, such as transforming the result from "one hot" results to some other complex results, not necessary to another device. Maybe we could call it `final_transform` or `result_transform`

Reply:
Hmm. One consideration: it is responsible for moving the data and target to the device. This is necessary because in some cases it chooses not to move the `target`, e.g. with Knowledge Distillation. Simply calling it a `final_transform` doesn't capture that requirement.

Reply:
`transform_to_device_fn`?